### PR TITLE
Remove default vnet and route table creation when using Azure CNI.

### PR DESCRIPTION
**Pull request by JunSun17**
Remove default vnet and route table creation when using Azure CNI.

Also correct the VNET/Subnet/Vnet resource group in output so they can be consumed by HCP later in its controller manger configuration.

This is the change in ACS-E side. Verified in manual test.

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@JunSun17 you might want to rebase to see the newly added unit test coverage diff

**Comment by jackfrancis** [comment]
What's the high level intention here?

**Comment by JunSun17** [comment]
@jackfrancis The intention here is when custom brings their own vnet/subnet, ACS-E should not create a default vnet/subnet.

**Comment by jackfrancis** [comment]
@JunSun17 so currently, there is an identified bug that an "extra" vnet is created and left empty in the resource group for "bring your own VNET" cluster creation scenarios?

Is this bug observed in Azure CNI only, or also for kubenet network plugin flows?

**Comment by JunSun17** [comment]
@jackfrancis Yes, it is an existing issue for both Azure CNI and kubenet when custom brings in their own vnet/subnet.

File: parts/k8s/kubernetesbase.t [hunk 1]
```diff
@@ -62,6 +62,7 @@
     {{if not IsHostedMaster}}
       ,{{template "k8s/kubernetesmasterresources.t" .}}
     {{else}}
+{{if not IsCustomVNET}}
```
[the remark this answers is not in the document]
Comment (JunSun17) [reply parts/k8s/kubernetesbase.t]: Sure, done.

**Comment by jackfrancis** [comment]
@JunSun17 FYI I rebased to fix master merge conflict

Will merge when we have an lgtm from AKS rep

**Comment by JunSun17** [comment]
@jackfrancis Thanks! I need to make more changes on this PR, will let you know when it is ready for merge.

**Comment by JunSun17** [comment]
Please do not merge this yet, will do more testing and update when it is ready to merge.

File: pkg/api/convertertoagentpoolonlyapi.go [hunk 2]
```diff
@@ -230,7 +230,7 @@ func convertV20170831AgentPoolOnlyOrchestratorProfile(kubernetesVersion string)
 			EnableRbac:          helpers.PointerToBool(false),
 			EnableSecureKubelet: helpers.PointerToBool(false),
 			// set network default for un-versioned model
-			NetworkPolicy:      string(v20180331.Kubenet),
+			NetworkPlugin:      string(v20180331.Kubenet),
```
Comment (weinong) [inline pkg/api/convertertoagentpoolonlyapi.go]: It seems to break backward compatibility on existing AKS which has `NetworkPolicy`

**Pull request merged**